### DISCUSSION TITLE
How can i get tradeId when we remove undertow plugin

### DISCUSSION THREAD

**Discussion by teamoTNT** [apache/skywalking#8908] [Q&A]
In SpringBoot, When undertow is used to replace tomcat and we remove undertow plugin.
Currently we use this method (org.apache.skywalking.apm.toolkit.trace.TraceContext.traceId()) to get tradeId to integrate logtube.
But we got 'N/A' instead of normal tradeId in log.
So, how can we get the correct tradeId.

**Comment by wu-sheng** [comment]
That method is reading trace ID from the current context. 

Removing the undertow plugin or not is irrelevant to your question. I am not sure why you thought it is.

**Reply by teamoTNT** [reply]
But when i not remove the ndertow plugin, it can get correct tradeId.

**Reply by wu-sheng** [reply]
Do you know what is tracing context and what is tracing? If not, it is hard to discuss this. The original issue is about duplicated information, it has nothing related to you. But you insist on keeping mentioning that.

**Reply by teamoTNT** [reply]
It may be a problem with the load order of the ServeletFilter,  When Logtube ServeletFilter load before Skywalking that logtube get 'N/A' not  correct trade Id from the current context.
Is there any information about undertow plugin? I cannot find anything but source code.

**Reply by wu-sheng** [reply]
I think I had confirmed there is no integration or test about `Logtube`.  I even can't find any official repository on GitHub.

https://github.com/logtube/logtube-java This is what I can get, but nothing likes a real open-source project.

Once you are using something we don't know, there is nothing we can do.
